### PR TITLE
Open vino

### PR DESCRIPTION
**Pull request by probicheaux**
# Description

Add in openvino to repo

openvino_640_seg
openvino_640
openvino_1280_seg
openvino_1280

## Type of change

Please delete options that are not relevant.

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] This change requires a documentation update

## How has this change been tested, please provide a testcase or example of how you tested the change?

locally

## Any specific deployment considerations

no

## Docs

-   [ ] Docs updated? What were the changes:

### CODE REVIEW

**Comment by paulguerrie** [comment]
Does `onnxruntime-openvino` work on arm based systems?  I know it won't accelerate, but does it still function?

File: inference/core/env.py [hunk 1]
```diff
@@ -224,7 +224,7 @@
 NUM_WORKERS = int(os.getenv("NUM_WORKERS", 1))
 
 ONNXRUNTIME_EXECUTION_PROVIDERS = os.getenv(
-    "ONNXRUNTIME_EXECUTION_PROVIDERS", "[CUDAExecutionProvider,CPUExecutionProvider]"
+    "ONNXRUNTIME_EXECUTION_PROVIDERS", "[CUDAExecutionProvider,OpenVINOExecutionProvider]"
```
Comment (paulguerrie) [inline inference/core/env.py:227]: Should we leave `CPUExecutionProvider` as the last one incase something with the OpenVINO dependancies don't work?

File: requirements/requirements.cpu.txt [hunk 2]
```diff
@@ -1,2 +1,2 @@
-onnxruntime<=1.15.1
+onnxruntime-openvino<=1.15.1
```
Comment (paulguerrie) [inline requirements/requirements.cpu.txt:1]: Does onnxruntime-openvino work on arm based systems? I know it won't accelerate, but does it still function?

**Comment by PawelPeczek-Roboflow** [comment]
Before accepting this - could you provide benchmark results describing impact of the change?

File: requirements/requirements.cpu.txt (hunk 2 above)
Comment (probicheaux) [reply requirements/requirements.cpu.txt:1]: the package doesnt even exist

**Review by PawelPeczek-Roboflow — APPROVED** [review]

File: docker/dockerfiles/Dockerfile.onnx.cpu [hunk 3]
```diff
@@ -43,13 +44,18 @@ RUN pip3 install --upgrade pip  && pip3 install \
...
+ARG TARGETPLATFORM
+
 COPY --from=base / /
 
 WORKDIR /build
 COPY . .
 RUN make create_wheels
 RUN pip3 install dist/inference_cli*.whl dist/inference_core*.whl dist/inference_cpu*.whl dist/inference_sdk*.whl
 
+RUN if [ "${TARGETPLATFORM}" = "linux/amd64" ]; then pip3 install -r requirements/requirements.vino.txt; rm -rf ~/.cache/pip; fi
```
Comment (paulguerrie) [inline docker/dockerfiles/Dockerfile.onnx.cpu:56]: Do we need to uninstall onnxruntime before we install requirements/requirements.vino.txt? Wondering if we'll have two conflicting versions of onnxruntime if we don't explicitly uninstall it first.

**Review by paulguerrie — APPROVED** [review]
One question. Otherwise approved!

File: docker/dockerfiles/Dockerfile.onnx.cpu (hunk 3 above)
Comment (PawelPeczek-Roboflow) [reply docker/dockerfiles/Dockerfile.onnx.cpu:56]: no, both are visible. confirmed manually

Comment (PawelPeczek-Roboflow) [reply docker/dockerfiles/Dockerfile.onnx.cpu:56]:
```
(base) ppeczek@pawel-dev:~/inference$ docker run -it --entrypoint bash roboflow/roboflow-inference-server-cpu:dev
root@e30729664123:/app# python
Python 3.9.18 (main, Nov 29 2023, 05:56:31)
[GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import onnxruntime
>>> onnxruntime.get_available_providers()
['OpenVINOExecutionProvider', 'CPUExecutionProvider']
```

**Pull request merged**